### PR TITLE
[5.2] Introduces `Hasher::isHash()` method

### PR DESCRIPTION
**Pull request by inxilpro**
`Hasher::isHash()` provides a simple way to confirm that a given string hash been hashed by the hasher. This may be useful for automatically hashing passwords and other sensitive data before save (that's my use case).

### CODE REVIEW

File: src/Illuminate/Contracts/Hashing/Hasher.php [hunk 1]
```diff
@@ -31,4 +31,14 @@ public function check($value, $hashedValue, array $options = []);
...
+
+    /**
+     * Check if the given string is a hash produced by this hasher.
+     *
+     * @param  string $value
+     * @param  array $options
+     * @param  bool $strict
+     * @return bool
+     */
+    public function isHash($value, array $options = [], $strict = false);
```
Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Hashing/Hasher.php]: Adding a method to a contract is a major breaking change.

File: src/Illuminate/Hashing/BcryptHasher.php [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 namespace Illuminate\Hashing;
 
-use RuntimeException;
 use Illuminate\Contracts\Hashing\Hasher as HasherContract;
+use RuntimeException;
```
Comment (GrahamCampbell) [inline src/Illuminate/Hashing/BcryptHasher.php]: Please don't change this order.

File: src/Illuminate/Hashing/BcryptHasher.php [hunk 3]
```diff
@@ -17,8 +17,8 @@ class BcryptHasher implements HasherContract
     /**
      * Hash the given value.
      *
-     * @param  string  $value
```
Comment (GrahamCampbell) [inline src/Illuminate/Hashing/BcryptHasher.php]: Please could you revert the cs changes to the phpdoc here, and in the other similar places. :)

**Comment by inxilpro** [comment]
I've fixed the code style issues I think. Should I close this PR and re-submit against the master branch?

**Comment by taylorotwell** [comment]
Going to hold off on this as I'm not sure all drivers would even have a reliable way of determining if an arbitrary string is a hash.

**Comment by inxilpro** [comment]
I was thinking that you could probably make a rudimentary `isHash` for just about anything (at the very least is it the right length and only contain allowed characters). But maybe a compromise would be to keep it out of the Interface and just implement `isHash` on the `BcryptHasher` (or expose some version of [`password_get_info`](https://secure.php.net/manual/en/function.password-get-info.php)).

The issue is that I want to confirm that a string has been hashed, and I'm concerned about something breaking in the future if I rely on `password_get_info` and assume that the underlying implementation stays the same.

The end goal is a way to automatically bcrypt certain values before saving them—but only if the value is not already hashed.

**Pull request closed without merging**